readthedocs/projects/views/public.py

```diff
@@ -90,7 +90,8 @@ def get_context_data(self, **kwargs):
             user=self.request.user, project=project)
 
         protocol = 'http'
-        if self.request.is_secure():
+        use_https = getattr(settings, 'PUBLIC_DOMAIN_USES_HTTPS', False)
```

Comment:
For this purpose, public domain != production domain. Badges are served from our production domain, docs are served from our public domain.

Reply:
So what about removing the scheme altogether?